src/scripts/clipperUI/ratingsHelper.ts

```diff
@@ -0,0 +1,538 @@
+import {Localization} from "../localization/localization";
+
+import {ClientType} from "../clientType";
+import {Constants} from "../constants";
+import {Utils} from "../utils";
+import {Settings} from "../settings";
+
+import {AnimationStrategy} from "./animations/animationStrategy";
+import {SlideFromRightAnimationStrategy} from "./animations/slideFromRightAnimationStrategy";
+
+import {SuccessPanelClass} from "./panels/successPanel";
+
+import {ClipperState} from "./clipperState";
+import {DialogButton} from "./panels/dialogPanel";
+import {Clipper} from "./frontEndGlobals";
+
+import * as Log from "../logging/log";
+
+export enum RatingsPromptStage {
+	NONE,
+	INIT,
+	RATE,
+	FEEDBACK,
+	END
+}
+
+interface RatingsLoggingInfo {
+	shouldShowRatingsPrompt: boolean;
+	badRatingTimingDelayIsOver?: boolean;
+	badRatingVersionDelayIsOver?: boolean;
+	clipSuccessDelayIsOver?: boolean;
+	doNotPromptRatings?: boolean;
+	lastBadRatingDate?: string;
+	lastBadRatingVersion?: string;
+	lastSeenVersion?: string;
+	numSuccessfulClips?: number;
+	ratingsPromptEnabledForClient?: boolean;
+	usedCachedValue?: boolean;
+}
+
+export class RatingsHelper {
+	public static rateUrlSettingNameSuffix = "_RatingUrl";
+	public static ratingsPromptEnabledSettingNameSuffix = "_RatingsEnabled";
+
+	/**
+	 * Get appropriate dialog panel message for the ratings prompt stage provided
+	 */
+	public static getMessage(stage: RatingsPromptStage): string {
+		switch (stage) {
+			case RatingsPromptStage.INIT:
+				return Localization.getLocalizedString("WebClipper.Label.Ratings.Message.Init");
+			case RatingsPromptStage.RATE:
+				return Localization.getLocalizedString("WebClipper.Label.Ratings.Message.Rate");
+			case RatingsPromptStage.FEEDBACK:
+				return Localization.getLocalizedString("WebClipper.Label.Ratings.Message.Feedback");
+			case RatingsPromptStage.END:
+				return Localization.getLocalizedString("WebClipper.Label.Ratings.Message.End");
+			default:
+			case RatingsPromptStage.NONE:
+				break;
+		}
+	}
+
+	/**
+	 * Get appropriate dialog panel buttons for the panel (with its internal states) provided
+	 */
+	public static getDialogButtons(panel: SuccessPanelClass): DialogButton[] {
+		let stage: RatingsPromptStage = panel.state.userSelectedRatingsPromptStage;
+		let clipperState: ClipperState = panel.props.clipperState;
+		let clientType: ClientType = clipperState.clientInfo.clipperType;
+
+		let buttons: DialogButton[] = [];
+
+		switch (stage) {
+			case RatingsPromptStage.INIT:
+				buttons.push({
```

Comment:
Probably good to refactor each of these cases into its own method for readability.